Jenkinsfile-utils

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env groovy
+
+/**
+ * noisePageBuild will create a build directory and compile and build the
+ * noisepage binary in that directory. The options passed into the method
+ * determine the compilation and build options. Refer to the defaultArgs
+ * for the different options that can be passed in.
+ */
+void noisePageBuild(Map args = [:]) {
+    Map defaultArgs = [
+        useCache: true,
+        buildType: 'Debug',
+        os: 'ubuntu',
+        isBuildTests: true,
+        useASAN: false,
+        isBuildBenchmarks: false,
+        isCodeCoverage: false,
+        isJumboTest: false,
+        isRecordTime: false
+    ]
+    Map config = defaultArgs << args
+    String compileCmd = generateCompileCmd(config)
+    String buildCmd = generateBuildCmd(config)
+    String buildScript = generateBuildScript(compileCmd, buildCmd, config.isRecordTime)
+
+    sh script:buildScript, label: 'Build'
+}
+
+/**
+ * generateCompileCmd creates the cmake command string. It is based on the
+ * config passed into the function. The config arguments are the same as the
+ * defaultArgs in noisePageBuild.
+ */
+String generateCompileCmd(Map config = [:]) {
+    String compileCmd = "cmake -GNinja -DCMAKE_BUILD_TYPE=$config.buildType"
+
+    if (config.isCodeCoverage) {
+        compileCmd += ' -DNOISEPAGE_GENERATE_COVERAGE=ON'
+    } else {
+        compileCmd += ' -DNOISEPAGE_UNITY_BUILD=ON'
```

Comment:
Nit: add a comment that UNITY_BUILD messes with the accurate tracking of code coverage.